mitodl/open-discussions
# Validate users' email address against mailgun

#### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5570

#### What's this PR do?
This adds management commands to upload a list of all users to mailgun to check for their validity. If an address is invalid that user is marked as inactive.

#### How should this be manually tested?
- Copy the `MAILGUN_*` settings from CI 
- Have some normal users created and at least one user with an invalid email (e.g. `example@invalid.domainname`).
- Run `./manage.py validate_user_emails start` to upload the data to be validated
- Run `./manage.py validate_user_emails check` and you should see that it has been uploaded. It might be in progress of if you only have a few users most likely already complete.
- Run `./manage.py validate_user_emails process` and then verify that the `EmailValidation` records have been updated with the results and invalid users have been marked inactive.

NOTE: the testing libraries around `requests` don't support mocking binary file downloads so I wasn't able to write tests for that portion of the code here. Since we're really only going to run this once and it isn't user facing, manual testing should be ok.
